aten/src/ATen/test/apply_test.cpp

```diff
@@ -1,121 +1,119 @@
-    CATCH_REQUIRE(ti.sizes[0] == (3 * 6));
-    CATCH_REQUIRE(ti.strides[0] == 22);
-    CATCH_REQUIRE(ti.sizes[1] == (5 * 2));
-    CATCH_REQUIRE(ti.strides[1] == 2);
+TEST(apply_test, Partial_Strided_Collapse_4D) {
+  int sizes[] = {3, 6, 5, 2};
+  int strides[] = {6 * 22, 22, 2 * 2, 2};
+  ::at::cuda::detail::TensorInfo<void, int> ti{nullptr, 4, sizes, strides};
+  ti.collapseDims();
+  EXPECT_EQ(ti.dims, 2);
+  EXPECT_EQ(ti.sizes[0], (3 * 6));
+  EXPECT_EQ(ti.strides[0], 22);
+  EXPECT_EQ(ti.sizes[1], (5 * 2));
+  EXPECT_EQ(ti.strides[1], 2);
 }
 
-CATCH_TEST_CASE("Collapsing Zeros and Ones", "Collapses a 5D tensor to a 1D tensor") {
-    int sizes[] = {1, 10, 1, 5, 4};
-    int strides[] = {4, 0, 16, 0, 1};
-    ::at::cuda::detail::TensorInfo<void, int> ti{nullptr, 5, sizes, strides};
-    ti.collapseDims();
-    CATCH_REQUIRE(ti.dims == 2);
-    CATCH_REQUIRE(ti.sizes[0] == (10 * 5));
-    CATCH_REQUIRE(ti.strides[0] == 0);
-    CATCH_REQUIRE(ti.sizes[1] == 4);
-    CATCH_REQUIRE(ti.strides[1] == 1);
+TEST(apply_test, Collapses_Zeros_and_Ones) {
+  int sizes[] = {1, 10, 1, 5, 4};
+  int strides[] = {4, 0, 16, 0, 1};
+  ::at::cuda::detail::TensorInfo<void, int> ti{nullptr, 5, sizes, strides};
+  ti.collapseDims();
+  EXPECT_EQ(ti.dims, 2);
+  EXPECT_EQ(ti.sizes[0], (10 * 5));
+  EXPECT_EQ(ti.strides[0], 0);
+  EXPECT_EQ(ti.sizes[1], 4);
+  EXPECT_EQ(ti.strides[1], 1);
 }
 
-CATCH_TEST_CASE("Collapsing to a Point Tensor", "Collapses a 3D tensor to a point tensor") {
-    int sizes[] = {1, 1, 1};
-    int strides[] = {17, 12, 3};
-    ::at::cuda::detail::TensorInfo<void, int> ti{nullptr, 3, sizes, strides};
-    CATCH_REQUIRE(ti.collapseDims() == 0);
-    CATCH_REQUIRE(ti.dims == 1);
-    CATCH_REQUIRE(ti.sizes[0] == 1);
-    CATCH_REQUIRE(ti.strides[0] == 1);
+TEST(apply_test, Collapse_to_Point_Tensor) {
+  int sizes[] = {1, 1, 1};
+  int strides[] = {17, 12, 3};
+  ::at::cuda::detail::TensorInfo<void, int> ti{nullptr, 3, sizes, strides};
+  EXPECT_EQ(ti.collapseDims(), 0);
+  EXPECT_EQ(ti.dims, 1);
+  EXPECT_EQ(ti.sizes[0], 1);
+  EXPECT_EQ(ti.strides[0], 1);
 }
 
-CATCH_TEST_CASE("Excluding in a 4D Contiguous", "Collapses a 4D tensor to a 3D tensor") {
-    int sizes[] = {3, 6, 5, 2};
-    int strides[] = {6 * 22, 22, 2 * 2, 2};
-    ::at::cuda::detail::TensorInfo<void, int> ti{nullptr, 4, sizes, strides};
-    CATCH_REQUIRE(ti.collapseDims(1) == 1);
-    CATCH_REQUIRE(ti.dims == 3);
-    CATCH_REQUIRE(ti.sizes[0] == 3);
-    CATCH_REQUIRE(ti.strides[0] == (6 * 22));
-    CATCH_REQUIRE(ti.sizes[1] == 6);
-    CATCH_REQUIRE(ti.strides[1] == 22);
-    CATCH_REQUIRE(ti.sizes[2] == (5 * 2));
-    CATCH_REQUIRE(ti.strides[2] == 2);
+TEST(apply_test, Excluding_in_Contiguous_4D) {
+  int sizes[] = {3, 6, 5, 2};
+  int strides[] = {6 * 22, 22, 2 * 2, 2};
+  ::at::cuda::detail::TensorInfo<void, int> ti{nullptr, 4, sizes, strides};
+  EXPECT_EQ(ti.collapseDims(1), 1);
+  EXPECT_EQ(ti.dims, 3);
+  EXPECT_EQ(ti.sizes[0], 3);
```

Comment:
Please do not use `EXPECT_*`. There is an important difference between `EXPECT_*` and `ASSERT_*`:
- When an `EXPECT_*`  fails, it will not fail the test. The test just continues. This is usually not desired.
- When an `ASSERT_*` fails, the whole test fails. This is much better  in most cases and is what Catch's `REQUIRE` also equates.

You will have to fix this throughout your PR.